docs/MemoryMap.md

```diff
@@ -0,0 +1,23 @@
+# Memory Map
+
+This file gives an overview of the flash layout used by the nRF52833 on the micro:bit v2.
```

Comment:
Worth putting in a tag or commit reference to when this is written. EG 'as of v2.whatever'